Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM gradle:latest as build
+
+WORKDIR /build
+
+COPY "gradle" "gradle"
+COPY "build.gradle" "settings.gradle" "gradle.properties" "/build"/
+
+RUN gradle build
+
+COPY src src
+
+RUN gradle build
+
+FROM openjdk:14
+
+WORKDIR /bot
+
+COPY --from=build /build/build/libs/bot-kt-*.jar /bot/bot-kt.jar
```

Comment:
This will keep old versions. Does docker redownload the Gradle cache every time or does it use a fresh clone?

Reply:
If the contents of the first container changes, then the second container will be remade, everything downstream of a change gets rebuilt.

Reply:
Yes, but does it delete the build folder or use a fresh clone?

Reply:
Anything that gets created on that step would be removed, so if step one is copy build dependency definition, step two is download dependencies, step three is copy source, step four build, etc... and source changes, then step 3 (and everything after/that depends on that step) gets rebuilt.